src/tools/auth0/handlers/connections.js

```diff
@@ -17,6 +18,50 @@ export const schema = {
   }
 };
 
+// addExcludedConnectionPropertiesToChanges superimposes excluded properties on the `options` object. The Auth0 API
+// will overwrite the options property when updating connections, so it is necessary to add excluded properties back in to prevent those excluded properties from being deleted.
+// This use case is common because organizations may not want to expose sensitive connection details, but want to preserve them in the tenant.
+// exported only for unit testing purposes
+export const addExcludedConnectionPropertiesToChanges = ({
+  proposedChanges,
+  existingConnections,
+  config
+}) => {
+  if (proposedChanges.update.length === 0) return proposedChanges;
+
+  const excludedFields = config()?.EXCLUDED_PROPS?.connections || [];
+  if (excludedFields.length === 0) return proposedChanges;
+
+  const newProposedUpdates = proposedChanges.update.map((proposedConnection) => {
+    const currConnection = existingConnections.find(({ id }) => id === proposedConnection.id);
+
+    if (currConnection === undefined) return proposedConnection;
+
+    // Only include fields that pertain to options
+    const currentExcludedPropertyValues = excludedFields.filter((excludedField) => excludedField.indexOf('options') === 0).reduce((agg, excludedField) => {
+      if (!dotProp.has(currConnection, excludedField)) return agg;
```

Comment:
Should you also check `dotProp.has(agg, excludedField);`?

Is there potentially a valid use case for `proposedChanges` to include excluded fields?

Reply:
I think the `dotProp.has(agg, excludedField);` would be checking for the case of duplicated excluded fields, which isn't a valid use case.

>Is there potentially a valid use case for proposedChanges to include excluded fields?

Not to my knowledge, otherwise folks would just unexclude to incorporate those changes.